File: src/qm/banking/domain/entities/Account.java
package qm.banking.domain.entities;

public abstract class Account {


    private final String IBAN;
    private int balance;

    public Account(String IBAN, int balance) {
        this.IBAN = IBAN;
        this.balance = balance;
    }

    public String getIBAN() {
        return IBAN;
    }

    public int getBalance() {
        return balance;
    }}

File: src/main/qm/banking/domain/entities/Account.kt
package qm.banking.domain.entities

abstract class Account(val iban: IBAN) {

    var balance = 0
        private set

    fun debit(amount: Int): Account {
        if (amount > balance) throw InsufficientFundsException(iban)
        balance -= amount
        return this
    }

    fun credit(amount: Int): Account {
        balance += amount
        return this
    }

}